classes/task/failed_task.php

```diff
@@ -31,6 +31,98 @@ public function get_name() {
         return get_string('taskfailed', 'logstore_xapi');
     }
 
+    private function extract_failed_events($limitnum) {
```

Comment:
Could you use protected instead of private?

Reply:
Could you add php doc blocks?

Reply:
Please declare default value if possible. In this case, 0 should be good.

Reply:
This was copied from the pre-existing emit_task.php